src/libs/Network/SequentialQueue.js

```diff
@@ -24,15 +40,38 @@ let isSequentialQueueRunning = false;
 function process() {
     const persistedRequests = PersistedRequests.getAll();
 
-    // This sanity check is also a recursion exit point
-    if (NetworkStore.isOffline() || _.isEmpty(persistedRequests)) {
+    // If we have no persisted requests or we are offline we don't want to make any requests so we return early
+    if (_.isEmpty(persistedRequests) || NetworkStore.isOffline()) {
+        requestThrottle.clear();
         return Promise.resolve();
     }
 
-    const task = _.reduce(persistedRequests, (previousRequest, request) => previousRequest.then(() => Request.processWithMiddleware(request, true)), Promise.resolve());
-
-    // Do a recursive call in case the queue is not empty after processing the current batch
-    return task.then(process);
+    // Get the first request in the queue and process it
+    const request = persistedRequests.shift();
+    return Request.processWithMiddleware(request, true).then(() => {
+        // If the request is successful we want to:
+        // - Remove it from the queue
+        // - Clear any wait time we may have added if it failed before
+        // - Call process again to process the other requests in the queue
+        PersistedRequests.remove(request);
+        requestThrottle.clear();
+        return process();
+    }).catch((error) => {
+        // If a request fails with a non-retryable error we just remove it from the queue and return
+        if (!_.contains(errorsToRetry, error.message)) {
+            PersistedRequests.remove(request);
+            return Promise.resolve();
+        }
+
+        // If the request failed and we want to retry it:
+        // - Check that we are not offline
+        // - Sleep for a period of time
+        // - Call process again. This will retry the same request since we have not removed it from the queue
+        if (NetworkStore.isOffline()) {
+            return Promise.resolve();
+        }
+        return requestThrottle.sleep(requestThrottle.getRequestWaitTime()).then(() => process());
```

Comment:
This also does not need to be returned. Once the catch block happens the promise is resolved and anything chained with it will run.